Sources/StreamChat/ChatClient.swift

```diff
@@ -58,7 +58,9 @@ public class ChatClient {
                 excludedUserIds: { [weak self] in Set([self?.currentUserId].compactMap { $0 }) },
                 emitEvent: { [weak center] in center?.process($0) }
             ),
-            ChannelReadUpdaterMiddleware(),
+            ChannelReadUpdaterMiddleware(
+                newProcessedMessageIds: { [weak center] in center?.newMessageIds }
```

Comment:
One small nit: we could move the messages ids optionality to here, and make `var newProcessedMessageIds: () -> Set<MessageId>` non optional
```suggestion
                newProcessedMessageIds: { [weak center] in center?.newMessageIds ?? [] }
```

Not a blocker for sure